src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -283,24 +280,25 @@ data class LightningOutgoingPayment(
     }
 
     sealed class Details {
+        abstract val paymentRequest: PaymentRequest
         abstract val paymentHash: ByteVector32
 
         /** A normal lightning payment. */
-        data class Normal(val paymentRequest: PaymentRequest) : Details() {
+        data class Normal(override val paymentRequest: Bolt11Invoice) : Details() {
             override val paymentHash: ByteVector32 = paymentRequest.paymentHash
         }
 
-        /** KeySend payments are spontaneous donations that don't need an invoice from the recipient. */
-        data class KeySend(val preimage: ByteVector32) : Details() {
```

Comment:
KeySend doesn't seem to be supported, should we just remove all references to it?

Reply:
True, we defined the types initially to ensure that the architecture allowed different types of lightning payments, but we don't support it. You can remove all references to it now that we have the `Blinded` case that shows a different type of lightning payment!